internal/elasticsearch/ingest/processors.go

```diff
@@ -0,0 +1,245 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package ingest
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"sort"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Processor represents an ingest processor.
+type Processor struct {
+	// Type of processor ("set", "script", etc.)
+	Type string `yaml:"-"`
+	// FirstLine is the line number where this processor definition starts
+	// in the pipeline source code.
+	FirstLine int `yaml:"-"`
+	// LastLine is the line number where this processor definitions end
+	// in the pipeline source code.
+	LastLine int `yaml:"-"`
+}
+
+// Processors return the list of processors in an ingest pipeline.
+func (p Pipeline) Processors() (procs []Processor, err error) {
+	switch p.Format {
+	case "yaml", "yml":
+		procs, err = processorsFromYAML(p.Content)
+	case "json":
+		procs, err = processorsFromJSON(p.Content)
+	default:
+		return nil, errors.Errorf("unsupported pipeline format: %s", p.Format)
+	}
+	return procs, errors.Wrapf(err, "failure processing %s pipeline '%s'", p.Format, p.Filename())
+}
+
+// extract a list of processors from a pipeline definition in YAML format.
+func processorsFromYAML(content []byte) (procs []Processor, err error) {
+	var p struct {
+		Processors []yaml.Node
+	}
+	if err = yaml.Unmarshal(content, &p); err != nil {
+		return nil, err
+	}
+	for idx, entry := range p.Processors {
+		if entry.Kind != yaml.MappingNode || len(entry.Content) != 2 {
```

Comment:
Is it always 2 or there is slight chance that the `Content` will contain something extra?

Reply:
My understanding its that is always 2 for MappingNode, but wanted to check just in case